pkg/registry/controller/strategy.go

```diff
@@ -92,9 +92,18 @@ func (rcStrategy) AllowCreateOnUpdate() bool {
 
 // ValidateUpdate is the default update validation for an end user.
 func (rcStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) field.ErrorList {
-	validationErrorList := validation.ValidateReplicationController(obj.(*api.ReplicationController))
-	updateErrorList := validation.ValidateReplicationControllerUpdate(obj.(*api.ReplicationController), old.(*api.ReplicationController))
-	return append(validationErrorList, updateErrorList...)
+	oldRc := old.(*api.ReplicationController)
+	newRc := obj.(*api.ReplicationController)
+
+	validationErrorList := validation.ValidateReplicationController(newRc)
+	updateErrorList := validation.ValidateReplicationControllerUpdate(newRc, oldRc)
+	errs := append(validationErrorList, updateErrorList...)
+
+	if len(api.NonConvertibleFields(oldRc.Annotations)) > 0 && !reflect.DeepEqual(oldRc.Spec.Selector, newRc.Spec.Selector) {
```

Comment:
Tighten up the conditional on the annotation you're expecting.  Then if you get an additional non-convertible field (one you don't recognize), fail.